src/d3a/models/market/blockchain_interface.py

```diff
@@ -199,12 +97,21 @@ def handle_blockchain_trade_event(self, offer, buyer, original_offer, residual_o
         return str(uuid.uuid4()), residual_offer
 
     def track_trade_event(self, trade):
-        path = Path.cwd().joinpath('src', 'd3a', 'models', 'market', 'metadata.json')
-        trade_id, call = self.storage_contract_call(
-            self.contracts['main'],
-            "trade",
-            str(path)
+
+        call_params = {
+            'id': 3,
+            'buyer': ALICE_STASH_ADDRESS,
+            'seller': BOB_STASH_ADDRESS,
```

Comment:
Ideally all agents should be having their own unique addresses unlike here where all agents are sharing the same address

Reply:
This is part of the DID epic and I am not sure we should do it in this ticket... Or do you suggest generating a random address for every participant?

Reply:
Hmm good idea, better post a link of this comment to DID epic in order for the first thing that needs to be changed while implementing DID

Reply:
Added https://gridsingularity.atlassian.net/browse/DD-12